rule.php

```diff
@@ -173,50 +102,55 @@ public static function add_settings_form_fields(mod_quiz_mod_form $quizform, Moo
     public static function validate_settings_form_fields(array $errors,
                                                          array $data, $files, mod_quiz_mod_form $quizform) : array {
         global $DB;
-        $settings = self::filter_plugin_settings((object) $data);
-
-        // Validate basic settings using persistent class.
-        $quizsettings = (new quiz_settings())->from_record($settings);
-        // Set non-form fields.
-        $quizsettings->set('quizid', $data['instance']);
-        $quizsettings->set('cmid', $data['coursemodule']);
-        $quizsettings->validate();
-
-        // Add any errors to list.
-        foreach ($quizsettings->get_errors() as $name => $error) {
-            $name = self::add_prefix($name); // Re-add prefix to match form element.
-            $errors[$name] = $error->out();
-        }
 
-        // If there have been any quiz attempts, prevent settings being saved.
-        if ($quizform->get_instance()) {
-            $attempts = $DB->get_records('quiz_attempts', ['quiz' => $quizform->get_instance()]);
-            if (!empty($attempts)) {
-                $errors['seb'] = get_string('settingsfrozen', 'quizaccess_seb');
-            }
-        }
+        if (settings_provider::can_configure_seb($quizform->get_context())) {
+            if (!settings_provider::is_seb_settings_locked($data['instance'])) {
+                $settings = settings_provider::filter_plugin_settings((object) $data);
+
+                // Validate basic settings using persistent class.
+                $quizsettings = (new quiz_settings())->from_record($settings);
+                // Set non-form fields.
+                $quizsettings->set('quizid', $data['instance']);
+                $quizsettings->set('cmid', $data['coursemodule']);
+                $quizsettings->validate();
+
+                // Add any errors to list.
+                foreach ($quizsettings->get_errors() as $name => $error) {
+                    $name = settings_provider::add_prefix($name); // Re-add prefix to match form element.
+                    $errors[$name] = $error->out();
+                }
 
-        // Edge case for filemanager_sebconfig.
-        $requiresetting = $quizsettings->get('requiresafeexambrowser');
-        if ($requiresetting == settings_provider::USE_SEB_UPLOAD_CONFIG) {
-            $itemid = $data['filemanager_sebconfigfile'];
-            $drafterror = settings_provider::validate_draftarea_configfile($itemid);
-            if (!empty($drafterror)) {
-                $errors['filemanager_sebconfigfile'] = $drafterror;
-            }
-        }
+                // If there have been any quiz attempts, prevent settings being saved.
+                if ($quizform->get_instance()) {
+                    $attempts = $DB->get_records('quiz_attempts', ['quiz' => $quizform->get_instance()]);
```

Comment:
This is counting the preview attempts if any, the settings page does not save.

Should we call is_seb_settings_locked() or the raw quiz_has_attempts() function.

Reply:
Or pass 'preview' => 0, which is what the function does.

Reply:
we use is_seb_settings_locked already. This code should go away. Will fix it.

Reply:
Fixed. + Add better warning message about licked settings.

Reply:
Nice warning notice.